### PR TITLE
Rework Introduction

### PR DESCRIPTION
**Pull request by fl3pp**

### CODE REVIEW

File: SA/design_concept/content/introduction/goals.typ [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#import "../../../acronyms.typ": *
+
+= Goals <goals>
+
+With VisualFP we want to design a tool that allows graphical development of
```
Comment (lstreckeisen) [inline SA/design_concept/content/introduction/goals.typ:5]: The goal of this term paper is to design a tool, VisualFP, that allows the graphical development of
functional code, is block-based and extensible.
The target audience of VisualFP are students learning to program.
Additionally, VisualFP should offer support for experienced programmers
that want to view their code in a visual context.

The core functionality of VisualFP will be implemented in a
#ac("PoC") to proof the feasibility of the design.

The main part of this paper is to transparently document the design process that was used
to create VisualFP. This allows others to better understand
the decisions that have been made, which alternatives were considered, and
to evaluate for themselves whether they want to follow the same path.

File: SA/design_concept/content/introduction/motivation.typ [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+not possible in such an environment. Examples include Scratch, developed by
+MIT, and LEGO Mindstorms software.
+
+There aren't any similar tools available to teach functional programming,
+which is why most educators start their courses by either showing the
+logical nature of functional programming through e.g. the lambda calculus,
+or by jumping right into code, leaving many students puzzled.
+
+Some visual approaches to functional programming do exist though, some of
+which are covered in @existing_tools. Unfortunately, none are not suitable
```
Comment (lstreckeisen) [inline SA/design_concept/content/introduction/motivation.typ:15]:
```suggestion
which are covered in @existing_tools. Unfortunately, none are suitable
```

File: SA/design_concept/content/introduction/motivation.typ [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+as introduction to general purpose functional programming.
```
Comment (lstreckeisen) [inline SA/design_concept/content/introduction/motivation.typ:16]:
```suggestion
as an introduction to general purpose functional programming.
```

File: SA/design_concept/content/introduction/motivation.typ [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+Block-based tools don't need to be limited to education: A sufficiently
```
Comment (lstreckeisen) [inline SA/design_concept/content/introduction/motivation.typ:18]:
```suggestion
However, block-based tools don't need to be limited to education: A sufficiently
```

File: SA/design_concept/content/introduction/tool_research.typ [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+= Existing tools <existing_tools>
+
+There already are tools available that could fill the role of a visual
+functional programming language as described in @motivation.
+
+But even if they fail to live up to the specific goals of this project, they
+may still be useful as inspiration or as a starting point for a VisualFP.
+
+In the following, we will take a look at some of these tools and discuss
```
Comment (lstreckeisen) [inline SA/design_concept/content/introduction/tool_research.typ:11]:
```suggestion
This section discusses these tools and
```

File: SA/design_concept/content/introduction/goals.typ (hunk 1 above)
Comment (fl3pp) [reply SA/design_concept/content/introduction/goals.typ]: Applied

**Review by lstreckeisen — APPROVED** [review]

**Pull request merged**